src/PrestaShopBundle/Security/Admin/EmployeeProvider.php

```diff
@@ -39,19 +44,34 @@
  */
 class EmployeeProvider implements UserProviderInterface
 {
-    const ROLE_EMPLOYEE = 'ROLE_EMPLOYEE';
-
+    /**
+     * @var Context
+     */
     private $legacyContext;
 
     /**
      * @var CacheItemPoolInterface
      */
     private $cache;
 
-    public function __construct(LegacyContext $context, CacheItemPoolInterface $cache)
-    {
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @param LegacyContext $context
+     * @param CacheItemPoolInterface $cache
+     * @param CommandBusInterface $queryBus
+     */
+    public function __construct(
+        LegacyContext $context,
+        CacheItemPoolInterface $cache,
+        CommandBusInterface $queryBus
```

Comment:
So this breaks BC as well :roll_eyes:

Reply:
@matks should I make a separate class and deprecate this one as well, to keep it backwards compatible?

Reply:
Yes indeed